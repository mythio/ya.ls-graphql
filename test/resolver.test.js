require("dotenv").config();
const gql = require("graphql-tag");
const jwt = require("jsonwebtoken");
const { createTestClient } = require("apollo-server-testing");

const db = require("./utils/db");
const User = require("../models/user");
const serverInit = require("./utils/server");
const ShortUrl = require("../models/shortUrl");

describe("Resolver", () => {
  beforeAll(db.connectToDB);
  afterAll(db.disconnectDB);
  afterEach(db.cleanDB);

  describe("query.me", () => {
    test("should have correct query", async () => {
      const user = new User({
        _id: "5e4dcdfcc76d441afd3d29d9",
        name: "user_name",
        emailAddress: "email@address.com",
        password: "123123123",
        shortIds: []
      });

      await user.save();

      const q = gql`
        query {
          me {
            userId
            name
            emailAddress
            shortIds {
              shortId
              originalUrl
            }
          }
        }
      `;

      const token = jwt.sign({ userId: user._id }, process.env.USER_SECRET);

      const server = serverInit({ authorization: token });

      const { query } = createTestClient(server);
      const res = await query({ query: q });
      expect(res.errors).toBeUndefined();
      expect(res.data).toMatchSnapshot();
    });
  });
});
