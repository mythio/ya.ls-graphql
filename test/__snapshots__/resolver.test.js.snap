// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resolver query.me should have correct query 1`] = `
Object {
  "me": Object {
    "emailAddress": "email@address.com",
    "name": "user_name",
    "shortIds": Array [],
    "userId": "5e4dcdfcc76d441afd3d29d9",
  },
}
`;
